### PR TITLE
Refresh the `README.md`

### PR DESCRIPTION
**Pull request by mamercad**
Fix the broken Slack badge image, remove contractions, make grammar changes, write OS names and versions more consistently, show long options in commands, etc. I'll add markdown linting ~in another PR~ shortly.

### CODE REVIEW

**Review by armab — CHANGES_REQUESTED** [review]
I'm :+1:  for the Readme refresh, but :-1: for Markdown linting as that's another uneeded check that might block a PR or change.

I think markdown linting makes sense for the projects where every file is documentation, but not for a single `README.md` when the frequency of change is low and the chance of breaking a Markdown file is even lower (they just work!).
Let's avoid utilizing computing resources in this case :green_apple:

**Comment by mamercad** [comment]
> > I'm 👍 for the Readme refresh, but 👎 for Markdown linting as that's another uneeded check that might block a PR or change.
> > I think markdown linting makes sense for the projects where every file is documentation, but not for a single `README.md` when the frequency of change is low and the chance of breaking a Markdown file is even lower (they just work!). Let's avoid utilizing computing resources in this case 🍏
> 
> Sure, I'll push without the workflow shortly.

Done, I rebased as well so the history should be clean.

**Comment by mamercad** [comment]
In the spirit of not running workflows unnecessarily, I'm noticing that the `build` workflow is running (for this `README.md` update, we should probably fix that).

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -11,7 +11,10 @@ on:
     tags:
       - v*
   pull_request:
-    type: [opened, reopened, edited]
+    types: [opened, reopened, edited, synchronize]
+    paths-ignore:
+      - .markdownlint.yml
+      - README.md
```
Comment (armab) [inline .github/workflows/build.yaml:17]: Nice! :+1:

**Review by armab — APPROVED** [review]
Thanks for the refresh!

**Pull request merged**